compiler/src/dotty/tools/dotc/transform/Splicer.scala

```diff
@@ -289,6 +289,24 @@ object Splicer {
         unexpectedTree(tree)
     }
 
+    private def interpretArgs(argss: List[List[Tree]], fnType: Type)(using Env): List[Object] = {
+      fnType match
+        case fnType: PolyType => interpretArgs(argss, fnType.resType)
+        case fnType: MethodType if fnType.isErasedMethod => interpretArgs(argss, fnType.resType)
+        case fnType: MethodType =>
+          val interprededHeadParams =
+            for (info, arg) <- fnType.paramInfos.lazyZip(argss.head) yield
+              info match
+                case _: ExprType => () => interpretTree(arg) // by-name argument
+                case _ => interpretTree(arg) // by-value argument
+          interprededHeadParams.toList ::: interpretArgs(argss.tail, fnType.resType)
+        case _ =>
+          // If the final result type is a context function the rest of the args are interpreted as by-value.
+          // for  `def f(x: Int): Int ?=> Int` we call directly the verison that does not create the closure
+          // `def f$direct(x: Int)(x$1: Int): Int`.
+          argss.flatMap(_.map(interpretTree))
```

Comment:
It's not clear how the comment is related to the actual code.

Reply:
The comment lets us know what those extra parameters are for and why we interpret them as by-value.

Reply:
It seems to be better to move the comment to the place where the logic is implemented. BTW, after #8386, there are no direct methods, IIRC.

Reply:
I rewrote the code to handle context functions explicitly and removed the comment.